stats/stats.go

```diff
@@ -203,24 +206,46 @@ func NewHistogram(Offset float64, Divider float64) *Histogram {
 	return h
 }
 
-// Tradeoff memory for speed (though that also kills the cache so...)
-// this creates an array of 100k (max value) entries
-// TODO: consider using an interval search for the last N big buckets
+// Val2Bucket values are kept in two different structure
+// val2Bucket allows you reach between 0 and 1000 in constant time
 func init() {
-	lastV := int32(lastValue)
-	val2Bucket = make([]int, lastV+1)
+	val2Bucket = make([]int, maxArrayValue)
+	for i, value := range histogramBucketValues {
```

Comment:
I'm a bit paranoid and the initialization to -1 is a bit far up; can you add
maxArrayValueIndex = -1
just before the for - ty